File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hm = {}
        hm[0] = 1

        res = 0
        ps = 0

        for num in nums:
            ps += num
            if ps - k in hm:
                res += hm[ps-k]
            
            hm[ps] = hm.get(ps, 0) + 1
    
        return res

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int res = 0;
        HashMap<Integer, Integer> hm = new HashMap<>();
        hm.put(0, 1);

        int ps = 0;
        for(int num: nums) {
            ps += num;
            if(hm.containsKey(ps - k)) {
                res += hm.get(ps-k);
            }
            hm.put(ps, hm.getOrDefault(ps, 0) + 1);
        }

        return res; 
    }
}